protocol/x/clob/keeper/process_operations.go

```diff
@@ -619,27 +619,43 @@ func (k Keeper) PersistMatchDeleveragingToState(
 	matchDeleveraging *types.MatchPerpetualDeleveraging,
 ) error {
 	liquidatedSubaccountId := matchDeleveraging.GetLiquidated()
+	perpetualId := matchDeleveraging.GetPerpetualId()
+	clobPair := k.mustGetClobPairForPerpetualId(ctx, perpetualId)
 
-	// Validate that the provided subaccount can be deleveraged.
-	if canDeleverageSubaccount, err := k.CanDeleverageSubaccount(ctx, liquidatedSubaccountId); err != nil {
-		panic(
-			fmt.Sprintf(
-				"PersistMatchDeleveragingToState: Failed to determine if subaccount can be deleveraged. "+
-					"SubaccountId %+v, error %+v",
-				liquidatedSubaccountId,
-				err,
-			),
-		)
-	} else if !canDeleverageSubaccount {
-		// TODO(CLOB-853): Add more verbose error logging about why deleveraging failed validation.
+	// If `IsFinalSettlement` flag on deleveraging match is set to true, verify that the market is in final settlement.
+	if matchDeleveraging.IsFinalSettlement && clobPair.Status != types.ClobPair_STATUS_FINAL_SETTLEMENT {
```

Comment:
This piece of validation is required so that block proposers cannot propose final settlement deleverage events for markets which are not in final settlement.